cpp/src/parquet/metadata.h

```diff
@@ -508,26 +508,29 @@ class PARQUET_EXPORT RowGroupMetaDataBuilder {
 
 class PARQUET_EXPORT FileMetaDataBuilder {
  public:
-  // API convenience to get a MetaData reader
+  // API convenience to get a MetaData builder
+  PARQUET_DEPRECATED("Deprecated. Use overload without KeyValueMetadata instead.")
```

Comment:
It's nice to say when it was deprecated.
```suggestion
  PARQUET_DEPRECATED("Deprecated in 12.0.0. Use overload without KeyValueMetadata instead.")
```